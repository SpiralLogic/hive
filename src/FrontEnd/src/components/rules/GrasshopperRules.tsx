import { FunctionComponent, h } from 'preact';
import Hexagon from '../Hexagon';
import Hextille from '../Hextille';
import Row from '../Row';
import RuleCell from './RuleCell';

export const GrasshopperRules: FunctionComponent = () => {
  return (
    <Hextille class="rules">
      <Row zIndex={1}>
        <Hexagon hidden={true} />
        <RuleCell result="correct" />
        <RuleCell creature="beetle" />
        <RuleCell
          creature="grasshopper"
          class="selected"
          correct={[
            ['left', 3],
            ['right', 3],
            ['bottomLeft', 5],
          ]}
          incorrect={[['bottomRight', 5]]}
        />
        <RuleCell creature="queen" zIndex={-1} />
        <RuleCell result="correct" />
      </Row>
      <Row>
        <Hexagon hidden={true} />
        <Hexagon hidden={true} />
        <RuleCell creature="grasshopper" />
        <Hexagon hidden={true} />
        <RuleCell creature="ant" />
      </Row>
      <Row>
        <Hexagon hidden={true} />
        <Hexagon hidden={true} />
        <RuleCell creature="queen" />
        <RuleCell creature="spider" />
        <RuleCell creature="ant" />
        <Hexagon hidden={true} />
      </Row>
      <Row>
        <RuleCell creature="spider" />
        <RuleCell result="correct" />
        <RuleCell creature="beetle" />
        <Hexagon hidden={true} />
        <RuleCell result="incorrect" />
      </Row>
      <Row>
        <Hexagon hidden={true} />
        <RuleCell creature="spider" />
        <RuleCell creature="spider" />
      </Row>
    </Hextille>
  );
};

GrasshopperRules.displayName = 'GrasshopperRules';
export default GrasshopperRules;
