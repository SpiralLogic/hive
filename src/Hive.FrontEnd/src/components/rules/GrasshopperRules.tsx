import Hexagon from '../Hexagon';
import Hextille from '../Hextille';
import Row from '../Row';
import RuleCell from './RuleCell';
import { Rule } from './rule';

const GrasshopperRules: Rule = {
  title: 'Grasshopper',
  description: ['Can just straight over other creatures', 'Can not jump gaps'],
  RuleComponent: () => (
    <Hextille class="rules">
      <Row zIndex={1}>
        <RuleCell result="correct" />
        <RuleCell creature="beetle" />
        <RuleCell
          creature="grasshopper"
          selected
          correctArrows={[
            ['left', 3],
            ['right', 3],
            ['bottomLeft', 5],
          ]}
          incorrectArrows={[['bottomRight', 5]]}
        />
        <RuleCell creature="queen" zIndex={-1} />
        <RuleCell result="correct" zIndex={-2} />
      </Row>
      <Row>
        <Hexagon hidden={true} />
        <Hexagon hidden={true} />
        <RuleCell creature="grasshopper" />
        <RuleCell />
        <RuleCell creature="ant" />
      </Row>
      <Row>
        <Hexagon hidden={true} />
        <RuleCell creature="queen" />
        <RuleCell creature="spider" />
        <RuleCell creature="ant" />
        <Hexagon hidden={true} />
      </Row>
      <Row>
        <Hexagon hidden={true} />
        <RuleCell result="correct" />
        <RuleCell creature="beetle" />
        <RuleCell />
        <RuleCell result="incorrect" />
      </Row>
      <Row>
        <Hexagon hidden={true} />
        <RuleCell creature="spider" />
        <RuleCell creature="spider" />
      </Row>
    </Hextille>
  ),
};
GrasshopperRules.RuleComponent.displayName = 'GrasshopperRules';
export default GrasshopperRules;
