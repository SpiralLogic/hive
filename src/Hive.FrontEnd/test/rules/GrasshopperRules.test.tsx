import { render } from '@testing-library/preact';
import GrasshopperRules from '../../src/components/rules/GrasshopperRules';

describe('<GrasshopperRules>', () => {
  it('snapshot', () => {
    const Rule = GrasshopperRules.Rule;
    expect(render(<Rule />).baseElement).toMatchSnapshot();
  });
});
