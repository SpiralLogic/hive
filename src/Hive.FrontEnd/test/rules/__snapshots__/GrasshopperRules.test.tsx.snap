// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grasshopper Rule snapshot tests snapshot 1`] = `
<body>
  <div>
    <div
      class="left"
      role="grid"
    >
      <div
        role="row"
        style="z-index: 1;"
      >
        <div
          class="correct"
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
            <span>
              ✓
            </span>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile beetle"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#beetle"
              />
            </svg>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile grasshopper selected"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#grasshopper"
              />
            </svg>
          </div>
          <svg
            class="arrow left correct"
            style="--span: 3;"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#arrow"
            />
          </svg>
          <svg
            class="arrow right correct"
            style="--span: 3;"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#arrow"
            />
          </svg>
          <svg
            class="arrow bottomLeft correct"
            style="--span: 5;"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#arrow"
            />
          </svg>
          <svg
            class="arrow bottomRight incorrect"
            style="--span: 5;"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#arrow"
            />
          </svg>
        </div>
        <div
          role="cell"
          style="z-index: -1;"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile queen"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#queen"
              />
            </svg>
          </div>
        </div>
        <div
          class="correct"
          role="cell"
          style="z-index: -2;"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
            <span>
              ✓
            </span>
          </div>
        </div>
      </div>
      <div
        role="row"
      >
        <div
          role="none"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
        </div>
        <div
          role="none"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile grasshopper"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#grasshopper"
              />
            </svg>
          </div>
        </div>
        <div
          class="blank"
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile ant"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#ant"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        role="row"
      >
        <div
          role="none"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile queen"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#queen"
              />
            </svg>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile spider"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#spider"
              />
            </svg>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile ant"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#ant"
              />
            </svg>
          </div>
        </div>
        <div
          role="none"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
        </div>
      </div>
      <div
        role="row"
      >
        <div
          role="none"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
        </div>
        <div
          class="correct"
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
            <span>
              ✓
            </span>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile beetle"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#beetle"
              />
            </svg>
          </div>
        </div>
        <div
          class="blank"
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
          </div>
        </div>
        <div
          class="incorrect"
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
            </svg>
            <span>
              ✘
            </span>
          </div>
        </div>
      </div>
      <div
        role="row"
      >
        <div
          role="none"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile spider"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#spider"
              />
            </svg>
          </div>
        </div>
        <div
          role="cell"
        >
          <svg
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              href="#hex"
            />
          </svg>
          <div
            class="tile spider"
          >
            <svg
              viewBox="0 0 100 100"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                href="#hex"
              />
              <use
                class="creature"
                href="#spider"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
